pharus/dynamic_api_gen.py

```diff
@@ -74,10 +94,14 @@ def {method_name}(jwt_payload: dict) -> dict:
             for grid in page["grids"].values():
                 if grid["type"] == "dynamic":
                     f.write(
-                        route_template.format(
+                        (
+                            route_template
+                            if values_yaml["SciViz"]["auth"]
+                            else route_template_nologin
```

Comment:
Rather than doing these repeated checks, we can simply check this before we start our `for` loop. Something to the effect of:

```python
resolved_route_template = route_template if values_yaml["SciViz"]["auth"] else route_template_nologin
```

Then we can use `resolved_route_template` as we did before with `route_template`.